dietpi/dietpi-software

```diff
@@ -3508,6 +3516,79 @@ _EOF_
 
 		fi
 
+		software_id=200 # DietPi Dashboard
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+
+			arch=
+			if (( $G_HW_ARCH == 10 )); then
+				arch='amd64'
+			elif (( $G_HW_ARCH == 3 )); then
+				arch='armv8'
+			elif (( $G_HW_ARCH == 2 )); then
+				arch='armv7'
+			else
+				arch='armv6'
+			fi
+
+			local version=$(sed -n '/^[[:blank:]]*SOFTWARE_DIETPI_DASHBOARD_VERSION=/{s/^[^=]*=//p;q}' /boot/dietpi.txt)
+			[[ $version ]] || version=Stable
+
+			G_WHIP_MENU_ARRAY=(
+					'Stable' ': Use a release version of the DietPi Dashboard.'
+					'Nightly' ': Use an unstable version of the DietPi Dashboard.'
+			)
+
+			local choice_required=
+			while :
```

Comment:
The loop needs to be removed then, the following is sufficient to leave the variable untouched when cancelled (or non-interactive):
```sh
G_WHIP_MENU "${choice_required}Please choose which version of DietPi Dashboard should be installed." && version=$G_WHIP_RETURNED_VALUE
```
Probably we could add:
```
G_WHIP_BUTTON_CANCEL_TEXT=$version
G_WHIP_DEFAULT_ITEM=$version
```
to pre-select the dietpi.txt value/default and replace the "Cancel" button text with the resulting selection when hitting it 🤔.